src/blockdata/transaction.rs

```diff
@@ -220,12 +220,73 @@ impl Default for TxIn {
         TxIn {
             previous_output: OutPoint::default(),
             script_sig: Script::new(),
-            sequence: u32::max_value(),
+            sequence: Sequence::default(),
             witness: Witness::default(),
         }
     }
 }
 
+/// Bitcoin transaction input sequencing number.
+/// 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Sequence(pub u32);
+
+impl Sequence {
+    const RBF_SIGNAL: u32 = 0xFFFFFFFE;
+    const DISABLE_FLAG_MASK: u32 = 0x80000000;
+    const LOCK_TYPE_MASK: u32 = 0x00400000;
+    
+    /// Retuns `true` if the sequence number is 0xffffffff.
+    pub fn is_final(&self) -> bool {
+        self.0 == MAX_SEQUENCE
+    }
+
+    /// Returns `true` if the sequence number is less than 0xfffffffe which
+    /// indicates that the transaction is opted-in to BIP125 replace-by-fee.
+    pub fn is_rbf(&self) -> bool {
+        self.0 < Sequence::RBF_SIGNAL
+    }
+
+    /// Returns `true` if the sequence has a relative timelock.
+    pub fn is_timelock(&self) -> bool {
+        self.0 & Sequence::DISABLE_FLAG_MASK == 0
+    }
+
+    /// Returns true if the sequence number encodes a block-based lock.
+    pub fn is_height_locked(&self) -> bool {
+        self.is_timelock() & (self.0 & Sequence::LOCK_TYPE_MASK == 0)
+    }
+
+    /// Returns true if the sequene number encodes a time based lock.
+    pub fn is_time_locked(&self) -> bool {
+        self.is_timelock() & (self.0 & Sequence::LOCK_TYPE_MASK > 0)
+    }
+    
+    /// Create a relative timelock using block height.
+    pub fn from_height(height: u16) -> Self {
+        Sequence::from(height as u32)
+    }
+
+    /// Create a relative timelock using time units where each unit is equivalent to 512 seconds.
+    pub fn from_timeunits(units: u16) -> Self {
+        Sequence::from(units as u32 | Sequence::LOCK_TYPE_MASK)
+    }
+}
+
+impl Default for Sequence {
+    fn default() -> Self {
+        Sequence(MAX_SEQUENCE)
+    }
+}
+
+impl From<u32> for Sequence {
```

Comment:
I think all the discussion on #994 about the naming of `to_consensus_u32` and `from_consensus` applies to sequence as well.

Reply:
I'm not a big fan of the naming but I've replaced `From<u32>` with the above two functions.

Reply:
@tcharding I actually think it doesn't. :) Sequence is a number, it's also publicly represented as a number, has no different imaginable encodings... The entire purpose of this newtype is to avoid mixing it with other integers. There's also implicit constructor `Sequence` just because the field is `pub`.

That being said, I dislike `From<u32> for Sequence`, `From<Sequence> for u32` is OK though. Also if people prefer consistency I'm fine with having `to_consensus`/`from_consensus`.

Reply:
I'll leave this one open for discussion. I personally think it's clearer to be using the Sequence constructor + `From<Sequence> for u32`.